File: go/go-http-server/shopping-list.go
package main

type ShoppingItem struct {
	Id   int
	Name string
}

type ShoppingList struct {
	Items map[int]ShoppingItem
}

func NewShoppingList() *ShoppingList {
	return &ShoppingList{
		Items: map[int]ShoppingItem{},
	}
}

func (s *ShoppingList) Add(item ShoppingItem) int {
	key := len(s.Items) + 1
	s.Items[key] = item
	return key
}

func (s *ShoppingList) Get(id int) ShoppingItem {
	return s.Items[id]
}

func (s *ShoppingList) List() map[int]ShoppingItem {
	return s.Items
}

File: com.damoo/src/main/java/com/damoo/repos/ShoppingList.java
package com.damoo.repos;

import java.util.ArrayList;
import com.damoo.models.ShoppingItem;
import org.springframework.stereotype.Repository;

@Repository
public class ShoppingList {

    private ArrayList<ShoppingItem> Items;

    public ShoppingList(){
        Items = new ArrayList<ShoppingItem>();
    }

    public void Add(ShoppingItem item){
        Items.add(item);
    }

    public ShoppingItem Get(int id){
        return Items.get(id);
    }

    public ArrayList<ShoppingItem> List(){
        return Items;
    }

}